dix/dispatch.c

```diff
@@ -2688,7 +2688,10 @@ ProcAllocColorCells(ClientPtr client)
         }
         nmasks = stuff->planes;
         length = ((long) npixels + (long) nmasks) * sizeof(Pixel);
-        Pixel *ppixels = calloc(1, length);
+
+        x_rpcbuf_t rpcbuf = { .swapped = client->swapped, .err_clear = TRUE };
+
+        Pixel *ppixels = x_rpcbuf_reserve(&rpcbuf, length);
```

Comment:
The `fee(ppixels)` below should be replaced with `x_rpcbuf_clear` (where `AllocColorCells` fails, sorry, GitHub doesn't allow me to add a comment under that line).